java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/SymbolicValue.java

```diff
@@ -105,25 +96,8 @@ public boolean references(SymbolicValue other) {
   }
 
   @Override
-  public boolean equals(Object o) {
-    if (this == o) {
-      return true;
-    }
-    if (o == null || getClass() != o.getClass()) {
-      return false;
-    }
-    SymbolicValue that = (SymbolicValue) o;
-    return id == that.id;
-  }
-
-  @Override
-  public int hashCode() {
-    return id;
-  }
-
-  @Override
   public String toString() {
-    return "SV_" + id;
+    return "SV_" + hashCode() % 100;
```

Comment:
Yes it is just to make it easier to read in debug outputs. This prints last two digits, should be enough for debug purposes.